R/DownloadModule.R

```diff
@@ -136,6 +136,11 @@ download_report_button_srv <- function(id,
           names(input_list) <- names(rmd_yaml_args)
           global_knitr <- list()
           if (is.logical(input$showrcode)) global_knitr <- list(echo = input$showrcode)
+          if (requireNamespace("formatR", quietly = TRUE)) {
+            global_knitr <- append(global_knitr, list(tidy.opts = list(width.cutoff = 60), tidy = TRUE))
+          } else {
+            message("For better code formatting, consider installing the formatR package.")
+          }
```

Comment:
Hey @kartikeyakirar the `global_knitr` used here is hidden from the user, and only used when `download`ed.
There are other ways of creating our report, and this happens directly through `render_and_compress`
https://github.com/insightsengineering/teal.reporter/blob/69b6e7ce93e4fe1c6f4fa3245e4a587cc99a5b9b/R/DownloadModule.R#L161

This `global_knitr` is also used in Previewer and Renderer. https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.reporter%20global_knitr&type=code

Let's unify the approach please. Can you change methods/functions that are exposed to a user ? https://github.com/insightsengineering/teal.reporter/issues/169#issuecomment-1733269638
Can you change the default values of all `global_knitr` parameters in all functions' declarations so it's visible something is setup by default, and not hidden inside the function?

Reply:
Got it. my intital impression was to fix the issues in the reports we downloaded. The renderRmd function generates the RMD file, and this adjustment can be made there.

> Can you change the default values of all global_knitr parameters in all functions' declarations so it's visible something is setup by default, and not hidden inside the function?

Yes, I can add the `global_knitr ` parameter to function declarations. However, I need to ensure that the user provides the correct parameter and that it's safely overwritten based on internal logic.

Reply:
jesus, this is the third time I need to say the same thing. I will actually show you with an arrow. Please change DEFAULT values of the function so it's exposed in the manual page of the function and visible for the user

![image](https://github.com/insightsengineering/teal.reporter/assets/133694481/9aa6dd18-65e3-4a75-b2dc-26f46a708453)

do not hardcode this stuff in the middle of the function as it's not clear for the person running the function that some hidden behavior is used

![image](https://github.com/insightsengineering/teal.reporter/assets/133694481/fe5b982d-246d-4d3a-a142-8d6c666fb66d)

Reply:
I changed it to draft mode while making updates. Sorry for confusion.